components/buildless-serverless/internal/controller/git/git_auth.go

```diff
@@ -0,0 +1,240 @@
+		return a.basicAuth()
+	default:
+		return nil, errors.New("unexpected authorization type")
+	}
+}
+
+func (a *GitAuth) GetAuthEnvs() []corev1.EnvVar {
+	s := a.secretName
+	var envs []corev1.EnvVar
+	envs = append(envs, corev1.EnvVar{
+		Name:  repositoryAuthTypeEnvVarName,
+		Value: string(a.authType),
+	})
+	envs = addEnvVar(envs, a.sshKey, s)
+	envs = addEnvVar(envs, a.username, s)
+	envs = addEnvVar(envs, a.password, s)
+	return envs
+}
+
+const (
+	kubernetesKeyFieldName         = "ssh-privatekey"
+	kubernetesUsernameFieldName    = "username"
+	kubernetesPasswordFieldName    = "password"
+	oldServerlessKeyFieldName      = "key"
+	oldServerlessUsernameFieldName = "username"
+	oldServerlessPasswordFieldName = "password"
+	repositoryAuthTypeEnvVarName   = "APP_REPOSITORY_AUTH_TYPE"
+	usernameEnvVarName             = "APP_REPOSITORY_USERNAME"
+	passwordEnvVarName             = "APP_REPOSITORY_PASSWORD"
+	sshKeyEnvVarName               = "APP_REPOSITORY_KEY"
+)
+
+func (a *GitAuth) parseSshAuthKubernetesSecret() error {
+	if a.authType != serverlessv1alpha2.RepositoryAuthSSHKey {
+		return errors.New(fmt.Sprintf("inconsistent secret type: %s, %s", a.authType, corev1.SecretTypeSSHAuth))
+	}
+	privateKey, ok := a.secret.Data[kubernetesKeyFieldName]
+	if !ok {
+		return errors.New(fmt.Sprintf("missing '%s'", kubernetesKeyFieldName))
+	}
+	a.sshKey = &dataField[[]byte]{
+		value:     privateKey,
+		fieldName: kubernetesKeyFieldName,
+		envName:   sshKeyEnvVarName,
+	}
+	return nil
+}
+
+func (a *GitAuth) parseBasicAuthKubernetesSecret() error {
+	if a.authType != serverlessv1alpha2.RepositoryAuthBasic {
+		return errors.New(fmt.Sprintf("inconsistent secret type: %s, %s", a.authType, corev1.SecretTypeBasicAuth))
+	}
+	username, usernameFound := a.secret.Data["username"]
+	password, passwordFound := a.secret.Data["password"]
+	if !usernameFound || !passwordFound {
+		return errors.New("missing 'username' or 'password'")
+	}
+	a.username = &dataField[string]{
+		value:     string(username),
+		fieldName: kubernetesUsernameFieldName,
+		envName:   usernameEnvVarName,
+	}
+	a.password = &dataField[string]{
+		value:     string(password),
+		fieldName: kubernetesPasswordFieldName,
+		envName:   passwordEnvVarName,
+	}
+	return nil
+}
+
+func (a *GitAuth) parseSshAuthOldServerlessSecret() error {
+	key, keyFound := a.secret.Data["key"]
+	if !keyFound {
+		return errors.New("missing 'key'")
+	}
+	a.sshKey = &dataField[[]byte]{
+		value:     key,
+		fieldName: oldServerlessKeyFieldName,
+		envName:   sshKeyEnvVarName,
+	}
+	password, passwordFound := a.secret.Data["password"]
+	if passwordFound {
+		a.password = &dataField[string]{
+			value:     string(password),
+			fieldName: oldServerlessPasswordFieldName,
+			envName:   passwordEnvVarName,
+		}
+	}
+	return nil
+}
+
+func (a *GitAuth) parseBasicAuthOldServerlessSecret() error {
+	username, usernameFound := a.secret.Data["username"]
+	password, passwordFound := a.secret.Data["password"]
```

Comment:
Why do you decide to not use the `oldServerlessUsernameFieldName` and `oldServerlessPasswordFieldName` variables here?